diff --git a/usr/sbin/rear b/usr/sbin/rear
index a51f820c..e21156bf 100755
--- a/usr/sbin/rear
+++ b/usr/sbin/rear
@@ -527,6 +527,10 @@ Build date: $( date -R )
 if test "$WORKFLOW" != "help" ; then
     # Create temporary work area and register removal exit task:
     BUILD_DIR="$( mktemp -d -t rear.XXXXXXXXXXXXXXX || Error "Could not create build area '$BUILD_DIR'" )"
+    # Since 'mktemp' doesn't always return a path under /tmp, the build
+    # directory has always to be excluded for safety
+    BACKUP_PROG_EXCLUDE+=( "$BUILD_DIR" )
+
     QuietAddExitTask cleanup_build_area_and_end_program
     Log "Using build area '$BUILD_DIR'"
     ROOTFS_DIR=$BUILD_DIR/rootfs
@@ -582,3 +586,5 @@ if test "$WORKFLOW" != "help" ; then
 fi
 
 exit $EXIT_CODE
+
+# vim: set et ts=4 sw=4:
diff --git a/usr/share/rear/conf/default.conf b/usr/share/rear/conf/default.conf
index ef118998..52b05eea 100644
--- a/usr/share/rear/conf/default.conf
+++ b/usr/share/rear/conf/default.conf
@@ -47,7 +47,9 @@
 # is set in the environment where /usr/sbin/rear is called.
 # To have a specific working area directory prefix for Relax-and-Recover
 # specify in /etc/rear/local.conf something like
+#
 #   export TMPDIR="/prefix/for/rear/working/directory"
+#
 # where /prefix/for/rear/working/directory must already exist.
 # This is useful for example when there is not sufficient free space
 # in /tmp or $TMPDIR for the ISO image or even the backup archive.
